pkg/manager/manager.go

```diff
@@ -335,11 +335,21 @@ func New(config *rest.Config, options Options) (Manager, error) {
 	}
 
 	// Create the resource lock to enable leader election)
-	leaderConfig := options.LeaderElectionConfig
-	if leaderConfig == nil {
+	var leaderConfig *rest.Config
+	var leaderRecorderProvider *intrec.Provider
+
+	if options.LeaderElectionConfig == nil {
 		leaderConfig = rest.CopyConfig(config)
+		leaderRecorderProvider = recorderProvider
+	} else {
+		leaderConfig = options.LeaderElectionConfig
```

Comment:
Please also copy the config here, the leader election manipulates it down the line and we don't want that to get propagated anywhere else

Reply:
I'm not sure what you mean. If given, that config would only be used for the leader election, so I don't see where manipulating it would be an issue.

I've updated the PR to copy it, since that's also not a problem. I'm just curious if you can provide some more information to help me learn.